### DISCUSSION TITLE
More reliable message receiving

### DISCUSSION THREAD

**Discussion by fherbert** [netmindz/balboa_GL_ML_spa_control#59] [Ideas]
I've had to tweak handleBytes a bit to get my setup working. I've changed the way my board now reads in the serial messages - only reading in serial messages meant for our panel by clearing the rx buffer when pin5 goes low and waiting until the correct number of bytes have been received, timing out if we haven't got what we expected. Been keen to hear if this change is something that the community would see as worthwhile.
https://github.com/fherbert/balboa_GL_ML_spa_control/commit/357fa76548cf7e7e94f890667c3f549ed52a4421

**Comment by netmindz** [comment]
Definitely looks improvement. Glad you managed to find how to clear from inside the interrupt and now we know there are only two message types and of fixed length (no such thing as FA long as that's really the FB from the other panel as reply) we always know how many bytes to read

**Comment by netmindz** [comment]
If you can create a PR I'll try and do some testing with both the regular and ESPHome codebases
